### ISSUE TITLE
Error using List<Int?>

### ISSUE THREAD

**Issue by fabioCollini** [square/moshi#633]
Using a `List<Int?>` in a Kotlin data class there is an error parsing a json using the generated adapter:

    Caused by: java.lang.IllegalArgumentException: Unexpected primitive int. Use the boxed type.

The error can be reproduced using this test:

    @JsonClass(generateAdapter = true)
    data class MyClass(
         val list: List<Int>,
         val nullableList: List<Int?>
    )
    
    class NullableIntListTest {
        @Test
        fun parsingTest() {
            val adapter = Moshi.Builder().build().adapter<MyClass>(MyClass::class.java)
            adapter.fromJson("""{
                |"list": [1, 2, 3],
                |"nullableList": [4, null, 5]
                |}""".trimMargin())
        }
    }

Looking at the adapter there is something strange:

    private val listOfIntAdapter: JsonAdapter<List<Int>> =
            moshi.adapter<List<Int>>(Types.newParameterizedType(List::class.java, Integer::class.java)).nonNull()

    private val listOfNullableIntAdapter: JsonAdapter<List<Int?>> =
            moshi.adapter<List<Int?>>(Types.newParameterizedType(List::class.java, Int::class.java)).nonNull()

Probably the second adapter should use an `Integer::class.java` instead of `Int::class.java`

Thanks for this great library!

**Comment by NightlyNexus** [comment]
The you for the test case!
Fixed in #627. This test passes on master.

**Comment by NightlyNexus** [comment]
This is now the generated code:
```kotlin
private val listOfIntAdapter: JsonAdapter<List<Int>> =
        moshi.adapter<List<Int>>(Types.newParameterizedType(List::class.java, Integer::class.java)).nonNull()

private val listOfNullableIntAdapter: JsonAdapter<List<Int?>> =
        moshi.adapter<List<Int?>>(Types.newParameterizedType(List::class.java, Int::class.javaObjectType)).nonNull()
```

**Comment by fabioCollini** [comment]
Thanks a lot @NightlyNexus !
Any plan to release a new version soon? Or is there a snapshot available?

**Comment by NightlyNexus** [comment]
Snapshot is available (1.7.0-SNAPSHOT) in the snapshots repository.
